src/platform/forms-system/test/js/fields/ObjectField.unit.spec.jsx

```diff
@@ -400,50 +397,106 @@ describe('Schemaform: ObjectField', () => {
     const uiSchema = {
       'ui:title': 'Blah',
     };
-    const idSchema1 = {
+    const idSchema = {
       $id: 'root',
       test: {
         $id: 'root_test',
       },
     };
-    const idSchema2 = {
+    const form = ReactTestUtils.renderIntoDocument(
+      <div>
+        <ObjectField
+          formContext={{ onReviewPage: true, pagePerItemIndex: 0 }}
+          uiSchema={uiSchema}
+          schema={schema}
+          idSchema={idSchema}
+          formData={{}}
+          onChange={onChange}
+          onBlur={onBlur}
+        />
+        <ObjectField
+          formContext={{ onReviewPage: true, pagePerItemIndex: 1 }}
+          uiSchema={uiSchema}
+          schema={schema}
+          idSchema={idSchema}
+          formData={{}}
+          onChange={onChange}
+          onBlur={onBlur}
+        />
+      </div>,
+    );
+    const formDOM = getFormDOM(form);
+    const ids = formDOM.querySelectorAll('legend');
+    expect(ids).to.have.length(2);
+    expect(ids[0].id).to.equal('root_test_0__title');
+    expect(ids[1].id).to.equal('root_test_1__title');
+  });
+  it('should render unique IDs for array items on review & submit page', () => {
+    // This occurs on form 526 when "ratedDisabilities" &
+    // "unempolyabilityDisabilities" both render the same list of rated
+    // disabilities
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const schema = {
+      type: 'object',
+      properties: {
+        test: {
+          type: 'boolean',
+        },
+      },
+    };
+    const uiSchema = {
+      'ui:title': 'Blah',
+      test: {
+        'ui:title': 'test',
+      },
+    };
+    const idSchema = {
       $id: 'root',
-      test2: {
-        $id: 'root_test2',
+      test: {
+        $id: 'root_test',
       },
     };
     const form = ReactTestUtils.renderIntoDocument(
       <div>
         <ObjectField
-          formContext={formContext}
+          formContext={{
+            onReviewPage: true,
+            pagePerItemIndex: 0,
+          }}
           uiSchema={uiSchema}
           schema={schema}
-          idSchema={idSchema1}
+          idSchema={idSchema}
           formData={{}}
           onChange={onChange}
           onBlur={onBlur}
         />
         <ObjectField
-          formContext={formContext}
+          formContext={{
+            onReviewPage: true,
+            pagePerItemIndex: 1,
+          }}
           uiSchema={uiSchema}
           schema={schema}
-          idSchema={idSchema2}
+          idSchema={idSchema}
           formData={{}}
           onChange={onChange}
           onBlur={onBlur}
         />
       </div>,
     );
     const formDOM = getFormDOM(form);
-    const ids = formDOM.querySelectorAll('legend');
+    const ids = formDOM.querySelectorAll('input');
```

Comment:
This and the previous test seem similar - is there a reason why `input` is used as a selector in one and `legend` is used in another?

Reply:
I meant to include both `legend` and `input` id checks. The previous tests is for non-unique IDs